sdk/samples/iot/pnp/pnp_protocol.c

```diff
@@ -124,7 +124,7 @@ static az_result is_component_in_model(
 {
   int32_t index = 0;
 
-  if (az_span_ptr(component_name) == NULL || az_span_size(component_name) == 0)
```

Comment:
I don't see a reason to check for both ptr == null and size == 0.

If ptr == NULL, size must already be 0 anyway.
If ptr != NULL, size may or may not be 0.

Therefore, checking if size == 0, covers all the possibilities.

Reply:
👍